src/server/list_family.cc

```diff
@@ -158,6 +158,22 @@ struct CircularMessages {
 // Used to recover logs for BLPOP failures. See OpBPop.
 thread_local CircularMessages debugMessages{50};
 
+// A bit awkward translation from a single key to ShardArgs.
+// We create a mutable slice (which will never be mutated) from the key, then we create
+// a CmdArgList of size 1 that references mslice and finally
+// we reference the first element in the CmdArgList via islice.
+struct SingleArg {
+  MutableSlice mslice;
+  IndexSlice islice{0, 1};
+
+  SingleArg(string_view arg) : mslice(const_cast<char*>(arg.data()), arg.size()) {
+  }
```

Comment:
that's tricky and only highlights to our bigger problem of exposing too much mutable slices

All `FinalizeWatched` does with args is this `for (string_view key : args) {`

Reply:
if I remember correctly `FinalizeWatched` is tricky because of BRPOPLPUSH